src/services/apiFaker.ts

```diff
@@ -0,0 +1,185 @@
+import * as faker from 'faker';
+
+import { randomShape } from "@app/services/randomShape";
+import { compare } from "@app/helpers/levenshtein";
+import { isObject, isNumber, objectFromEntries, randomElementFromArray } from '@app/helpers/helpers';
+import { keyMapObject, permittedFakerMethods, fakerExtension } from "@app/helpers/assets";
+
+
+const allKeys: keyMapObject[] = permittedFakerMethods
+    .map((permKey) => Object.keys(faker[permKey])
+    .map((fakerMethod) => ({ name: `${permKey}.${fakerMethod}`, key: permKey, value: fakerMethod })))
+    .reduce((a, b) => [...a, ...b]).concat(fakerExtension);
+
+
+export function createResponse(parsedReq: string): string {
+    let preparedData: object = JSON.parse(parsedReq);
+    preparedData = passDecode(preparedData, fakeValue, 'static');
+    preparedData = passRepeat(preparedData);
+
+    if (!Array.isArray(preparedData)) {
+        preparedData = passSettingsAndKeyDirective(preparedData);
+        if (typeof preparedData['@settings']['root'] !== 'undefined'
```

Comment:
Property `@settings` is an optional property. Accessing `@settings` property without checking is unsafe, as it can be undefined.